functions/knn.py

```diff
@@ -0,0 +1,61 @@
+pip ; install ; scikit-learn ; matplotlib ; opencv-python
+import os
+import cv2
```

Comment:
Reading GIF images should also be possible with skimage (scikit-image). Have you tried?

It's okay to use cv2 for this, but it's better to use skimage or matplotlib if possible, because cv2 is a pretty heavy dependency.